docs/edgeql/overview.rst

```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
        the results of the invocations are collected to form a single set.
 
 6. Collect the results of all iterations to obtain the final result set.
+
+
+.. _ref_eql_polymorphic_queries:
+
+Polymorphic Queries
+===================
+
+:index: nested shapes
+
+Sometimes it is useful to have an abstract type as a link target, thus
+allowing objects of different extending types to be referenced.  This
+necessitates writing *polymorphic queries* that could fetch different
+data depending on the type of the actual objects.  Consider the
+following schema:
+
+.. code-block:: sdl
+
+    abstract type Named {
+        required property name -> str {
+            delegated constraint exclusive;
+        }
+    }
+
+    type User extending Named {
+        property avatar -> str;
+        multi link favorites -> Named;
+    }
+
+    type Game extending Named {
+        property price -> int64;
+    }
+
+    type Article extending Named {
+        property url -> str;
+    }
+
+The above schema describes some platform where there are users, games
```

Comment:
"some platform" reads weird. I'd drop this sentence. The next one `Each User can add any of these entities` can be edited to "Every User`can have its `favorites` link point to either other User, Game, or Article"

Reply:
OK